addons/mail/models/mail_mail.py

```diff
@@ -427,13 +445,25 @@ def _prepare_outgoing_list(self, recipients_follower_status=None):
                 link_ids = {int(link) for link in re.findall(r'/web/(?:content|image)/([0-9]+)', body)}
                 if link_ids:
                     attachments = attachments - self.env['ir.attachment'].browse(list(link_ids))
+
+            # Compute email size and transform attachment in downloadable link in the body if it exceeds the limit
+            estimated_email_size = self._estimated_email_size(headers, body, [a.file_size for a in attachments.sudo()])
+            max_email_size_in_bytes = ((mail_server and mail_server.max_email_size) or
+                                       self.env['ir.mail_server']._get_default_max_email_size()) * 1024 * 1024
+            if estimated_email_size > max_email_size_in_bytes:
+                # Remove attachments and prepare downloadable link to be added in the body
+                attachments.generate_access_token()
+                attachments_links = self.env['ir.qweb']._render('mail.mail_attachment_links',
+                                                                {'attachments': attachments})
```

Comment:
Note that other calls to attachments are done using sudo... wondering why, because normally either the cron is sending emails (which means root), either a standard user (and access should be checked probably). Don't remember why the sudo is used in this method, but seems weird to have part of the code in sudo and part not in sudo.